scripts/umbrel-os/external-storage/monitor

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Load monitor-lib
+check_dependencies () {
+  for cmd in "$@"; do
+    if ! command -v $cmd >/dev/null 2>&1; then
+      echo "This script requires \"${cmd}\" to be installed"
+      exit 1
+    fi
+  done
+}
+
+check_dependencies readlink dirname
+UMBREL_ROOT="$(dirname $(readlink -f "${BASH_SOURCE[0]}"))/../../.."
+
+source "${UMBREL_ROOT}/scripts/monitor-lib"
+
+
+# Monitor external mount
+block_device="${1}"
+MOUNT_POINT="${2}"
+
+check_mount () {
+    # Check if Umbrel root is still bind mounted
+    if ! df -h "${UMBREL_ROOT}" | grep --quiet '/dev/sd'; then
+        echo "External device not bind mounted to Umbrel root"
+        return 1
+    fi
+
+    # Check if device is still mounted
+    if ! grep -q "${MOUNT_POINT}" /proc/mounts; then
+        echo "External device not mounted at ${MOUNT_POINT}"
+        return 1
+    fi
+
+    # Check if device is still registered
+    if ! cat /sys/block/${block_device}/ro; then
```

Comment:
Maybe `[[ ! -f "/sys/block/${block_device}/ro" ]]` would be a more appropriate check here.

Reply:
Ah thanks! Much better way to do that check :+1: